### PR TITLE
Implement `latest`  and `earliest` methods for a queryset

### PR DESCRIPTION
**Pull request by ruitcatarino**
## Description

This implements the latest (https://docs.djangoproject.com/en/5.1/ref/models/querysets/#latest) and earliest (https://docs.djangoproject.com/en/5.1/ref/models/querysets/#earliest) methods similar to Django.

## Motivation and Context

Like I explained in the issue #1753  I miss these two methods as someone coming from Django.

## How Has This Been Tested?

Well I implemented tests to validate the usage of both and ran them with `make test`. I also did some local testing.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added the changelog accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

The existing tests are not passing in the `develop` branch for me. But the new created tests are passing. I'll post the output of my tests here
```
===================================================================================== short test summary info ======================================================================================
FAILED tests/test_queryset.py::TestQueryset::test_delete - tortoise.exceptions.OperationalError: near "ORDER": syntax error
FAILED tests/test_queryset.py::TestQueryset::test_delete_limit - tortoise.exceptions.OperationalError: near "LIMIT": syntax error
FAILED tests/test_queryset.py::TestQueryset::test_delete_limit_order_by - tortoise.exceptions.OperationalError: near "ORDER": syntax error
FAILED tests/test_update.py::TestUpdate::test_update_with_limit_ordering - tortoise.exceptions.OperationalError: near "ORDER": syntax error
ERROR tests/test_default.py
ERROR tests/test_two_databases.py
4 failed, 1134 passed, 67 skipped, 4 xfailed, 2 errors in 36.45s
```

### CODE REVIEW

File: tortoise/queryset.py [hunk 1]
```diff
@@ -450,6 +450,38 @@ def order_by(self, *orderings: str) -> "QuerySet[MODEL]":
...
+        return queryset  # type: ignore
+
+    def earliest(self, field_name: str = "id") -> "QuerySetSingle[Optional[MODEL]]":
+        """
+        Returns the earliest object by ordering ascending on the specified field.
+
+        :params field_name: The field name to order by.
+        """
+        queryset = self._clone()
+        if not (
```
Comment (abondar) [inline tortoise/queryset.py:476]: Can you please add few test about passing related fields there? 
Like `related_model__pub_date`

I understand that it is not a prime usecase for that, but if we are supporting that behaviour - would be nice to have it covered

Comment (ruitcatarino) [reply tortoise/queryset.py]: I've added more tests to increase coverage, including testing the use of related fields `tests/test_latest_earliest.py`

**Review by abondar — APPROVED** [review]

**Comment by Abdeldjalil-H** [comment]
I think we should document these new methods.

**Comment by abondar** [comment]
They have automatic documentation
https://tortoise.github.io/query.html#tortoise.queryset.QuerySet.latest

Or you mean something more than that?

**Pull request merged**